### DISCUSSION TITLE
Add multiple executables into the same program?

### DISCUSSION THREAD

**Discussion by sad-dev** [NationalSecurityAgency/ghidra#5578] [Q&A]
As far as I can tell, the ``Add To Program`` function does not allow you to import a binary as executable (i.e. using the ELF/PE/etc. loaders) if the current program is already executable. I found myself wanting something like this on systems without ASLR e.g. a bootloader which loads several ELFs at hardcoded offsets.

Does anyone else have similar use cases? I settled for writing scripts to load raw binaries and place them correctly, but would love to be able to piggyback on the existing loaders in Ghidra.

**Comment by d-millar** [comment]
Possibly overkill, but the Windows core dump loader provides an example of how to add multiple PEs in a single program.

**Comment by ryanmkurtz** [comment]
There is support for this in the `Loader` interface, but as you noted we only provide it for the `BinaryLoader` and maybe a couple other similar ones.  Relocation support will be an issue for the mainstream loaders...ELF is the only one that lets you choose a new imagebase at time of import.  Then there is updating external references correctly, and probably other things too.  So, definitely non-trivial.

My sense is that this feature was intended to add memory dumps to an existing program, but it predates me by many years.
